src/components/NotesMindMap/NotesMindMapReducer.js

```diff
@@ -22,64 +21,73 @@ const defaultState = {
   noteText: '',
   isChangeParentModeActive: false,
   rootNote: undefined,
+  notes: [],
+  edges: []
 };
 
 export const notesMindMapReducer = (state = defaultState, { type, data }) => {
   const handleSelectedNoteChildrenFetchedAction = () => {
       const newState = cloneTreeInState();
       const childNotes = data;
+
       if (childNotes.length) {
-        newState.selectedNote.children = childNotes;
-      } else {
-        newState.selectedNote.hasNoChildren = true;
+        childNotes.forEach(child => {
+          newState.notes.push({ id: child.id, name: child.name, label: child.name, isNote: child.isNote /*group: (hasChildren ? 'parent' : 'children')*/ });
```

Comment:
Why not push the whole child? `newState.notes.push(child)` ?

Reply:
Previously it was
```
  newState.selectedNote.children = childNotes;
```
We just receive the children array and save it to the state. Without changing any properties for each node.
So why do we now need to change the properties for each node?

Reply:
The reason for this is to store the directed graph as arrays of nodes & edges. The childNotes are pushed to the nodes array & the relationships are in edges. If we have childNotes inside the node then that defeats the whole purpose of the transition.

Reply:
I mean why can't we change
```
 childNotes.forEach(child => {
          newState.notes.push({ id: child.id, name: child.name, label: child.name, isNote: child.isNote /*group: (hasChildren ? 'parent' : 'children')*/ });
```
To 
```
childNotes.forEach(child => newState.notes.push(child))
```
?